php/Block.php

```diff
@@ -62,29 +62,51 @@ public function register_block() {
 	 * @return string The markup of the block.
 	 */
 	public function render_callback( $attributes, $content, $block ) {
+		global $post;
+
 		$post_types = get_post_types( [ 'public' => true ] );
-		$class_name = $attributes['className'];
+		$attributes = wp_parse_args( $attributes, [ 'classname' => '' ] );
 		ob_start();
 
 		?>
-		<div class="<?php echo $class_name; ?>">
-			<h2>Post Counts</h2>
+		<div class="<?php echo esc_attr( $attributes['classname'] ); ?>">
+			<h2><?php esc_html__( 'Post Counts', 'site-counts' ); ?></h2>
 			<?php
 			foreach ( $post_types as $post_type_slug ) :
 				$post_type_object = get_post_type_object( $post_type_slug );
-				$post_count = count(
+				$post_count       = count(
 					get_posts(
 						[
-							'post_type' => $post_type_slug,
+							'post_type'      => $post_type_slug,
 							'posts_per_page' => -1,
 						]
 					)
 				);
 
 				?>
-				<p><?php echo 'There are ' . $post_count . ' ' . $post_type_object->labels->name . '.'; ?></p>
+				<p>
+					<?php
+					echo sprintf(
+						/* translators: 1: Post Count 2: Post Type */
+						esc_html__( 'There are %1$d %2$s', 'site-counts' ),
+						$post_count,
+						$post_type_object->labels->name
+					);
+					?>
+				</p>
 			<?php endforeach; ?>
-			<p><?php echo 'The current post ID is ' . $_GET['post_id'] . '.'; ?></p>
+
+			<?php if ( isset( $post ) ) : ?>
```

Comment:
Is there a way we could do this without relying on the global?

Reply:
> Is there a way we could do this without relying on the global?

I think relying on global is good, Because WordPress itself relying on global $post variable instead of `$_GET` . So global $post is always set on frontend and backend.